### DISCUSSION TITLE
Easiest way to enable sorting of a `TableView`

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#2645] [Q&A]
I sorta assumed `TableView` had built-in support for clicking on a header and having the table sort. Then I looked at `TableEditor` and saw it isn't so easy.

If I want to use 

```cs
jumpList.Table = new EnumerableTableSource<UnicodeRange> (UnicodeRange.Ranges, new Dictionary<string, Func<UnicodeRange, object>> () {
			{ "Category", (s) => s.Category },
			{ "Range", (s) => ($"(U+{s.Start:x5}-{s.End:x5})") },
		});
```

Is it possible/easy to enable changing which column the table is sorted by?

Tag @tznind

**Comment by tznind** [comment]
I have thought about this before and agree sorting support should be added.

But it needs to be optional as sorting very large tables can break
performance.

I think a good approach would be to add

SupportsSorting and Sort methods to ITableSource.  Then TableView can
handle the UI bits (clicks and sort indicators)

Will have a look and implement a good solution.

**Reply by tig** [reply]
In my case, after posting this I found it easier than expected. 

I would like auto sort indicators on column headers and a way of asking sort direction (ascending/descending) so I don't have to track.
